src/test/java/gov/cancer/tests/cts/advanced_search_form/CancerTypeSection_Test.java

```diff
@@ -0,0 +1,606 @@
+      SearchNavigationResult search_result = page.clickSubmit(page.getFormAction().getFindTrialsButton());
+      //create a comparator object from the redirected URL query part
+      QueryParametersComparator querycomp = new QueryParametersComparator(search_result.getPageURL().getQuery());
+      //Assert that url contains expected query
+      Assert.assertTrue(querycomp.compareQueryParams(expectedQueryParams ),
+        " query parameters does not match for Cancer Type, Stage and Side effects");
+    });
+  }
+  /** Verify that when user inputs a value to the Primary Cancer Type/Condition,SubType, Stage and Side Effects Fields
+   * and clicks Find Trials button, correct parameters are passed.
+   * parameters
+   * @param path       url
+   * @param cancerType cancer type value
+   * @param subType  cancer sub type value
+   * @param stage stage value
+   * @param sideEffect side effect value
+   * @param expectedQueryParams  list of query parameters
+   */
+  @Test(dataProvider = "getCancerTypeConditionSubTypeStageSideEffects")
+  public void verifyPrimaryCancerTypeSubTypeStageSideEffectsField(String path, String cancerType,String subType,String stage,
+                                                                  String sideEffect, String expectedQueryParams ) {
+    TestRunner.run(AdvancedSearchPage.class, path, (AdvancedSearchPage page) -> {
+      page.getCancerTypeSection().scrollUntilAllButtonVisible();
+      //Click All button
+      page.getCancerTypeSection().getAllButton().click();
+      //selecting the Cancer Type
+      page.getCancerTypeSection().getCancerTypeCondition().selectItem(cancerType);
+      //selecting the Sub Type
+      page.getCancerTypeSection().getSubType().selectItem(subType);
+      //selecting the Stage
+      page.getCancerTypeSection().getStage().selectItem(stage);
+      //selecting the Side Effect
+      page.getCancerTypeSection().getSideEffects().selectItem(sideEffect);
+
+      // create a navigation event redirect to search results page
+      SearchNavigationResult search_result = page.clickSubmit(page.getFormAction().getFindTrialsButton());
+      //create a comparator object from the redirected URL query part
+      QueryParametersComparator querycomp = new QueryParametersComparator(search_result.getPageURL().getQuery());
+      //Assert that url contains expected query
+      Assert.assertTrue(querycomp.compareQueryParams(expectedQueryParams ),
+        " query parameters does not match for Cancer Type, SubType, Stage and Side effects");
+    });
+  }
+
+  /** Verify that when user inputs an invalid value to the Primary Cancer Type/Condition field,
+   * error message is displayed.
+   * parameters
+   * @param path       url
+   * @param cancerType cancer type value
+   * @param error error message
+   */
+  @Test(dataProvider = "getInvalidCancerType")
+  public void verifyInvalidCancerTypeEntries(String path, String cancerType, String error) {
```

Comment:
Instead of `error`, naming it `expectedMessage` would be clearer to what it is.

Reply:
fixed it